oxide-auth/src/code_grant/accesstoken.rs

```diff
@@ -644,6 +645,11 @@ impl ErrorDescription {
 }
 
 impl BearerToken {
+    /// Given token parameters and a scope(s), create a new BearerToken.
+    pub fn new(token: IssuedToken, scope: String) -> BearerToken {
+        Self(token, scope)
+    }
```

Comment:
That should definitely not be the public signature, consider making the second parameter `&Scope`. But it's slightly controversial either way. This type consciously does not have a public constructor to nudge you towards properly using the flow and not issuing a badly protected token (e.g. to unauthenticated parties). I'll need to consider this, maybe you have arguments pro/con?

How complicated is it to duplicate? Seriously consider this alternative as well.